src/masternodes/mn_checks.cpp

```diff
@@ -2171,6 +2180,164 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor {
         return Res::Ok();
     }
 
+    
+    Res operator()(const CDUSDLockMessage& obj) const {
+        auto res = CheckCustomTx();
+        if (!res)
+            return res;
+
+        if (!HasAuth(obj.owner)) {
+            return Res::Err("Transaction must have at least one input from owner");
+        }
+
+        const auto attributes = mnview.GetAttributes();
+        if (!attributes) {
+            return Res::Err("Attributes unavailable");
+        }
+
+        const auto paramID = ParamIDs::DFIP2211D;
+
+        CDataStructureV0 activeKey{AttributeTypes::Param, paramID, DFIPKeys::Active};
+        CDataStructureV0 limitKey{AttributeTypes::Param, paramID, DFIPKeys::Limit, obj.batchId};
+        CDataStructureV0 tokenKey{AttributeTypes::Param, paramID, DFIPKeys::LockToken, obj.batchId};
+        if (!attributes->GetValue(activeKey, false)){
+            return Res::Err("DFIP2211D not currently active");
+
+        }
+        if(!attributes->CheckKey(limitKey) ||
+            !attributes->CheckKey(tokenKey)) {
+            return Res::Err("This batch is currently not available");
+        }
+
+        auto dusdToken = mnview.GetToken("DUSD");
+        if(!dusdToken) {
+            return Res::Err("Could not find dusd token");
+        }
+
+
+        auto tokenId= attributes->GetValue(tokenKey, (uint32_t)0);
+        auto limit = attributes->GetValue(limitKey,(uint32_t)0)*COIN;
```

Comment:
When you change limit key to `VerifyPositiveFloat` then remove `* COIN`

Reply:
tried that, doesn't work. question is how we want the limit to be put in. as fi or as whole DUSD. so should "limit = 10k DUSD" lead to a gov-value of 10k or 10k*1e8?
imho it should be "10000".
Then I need the *COIN here to compare limit with resultedMint (which is in fi) later on.

Reply:
For all amounts that we stored in attributes we used `VerifyPositiveFloat` which will convert 1.23456789 into int64 123456789 - so into COIN precision. You can use `VerifyPositiveFloatOrMinusOne` which was used on Consortium for mintLimit and mintLimitDaily which is probably what we need here as stating -1 is unlimited.

For consistency I would use this format, even if the limit will be huge here ant satoshis won't matter but we should keep it constant.